src/Avalonia.Controls/Flyouts/PopupFlyoutBase.cs

```diff
@@ -424,9 +441,7 @@ private void PositionPopup(bool showAtPointer)
             else
             {
                 Popup.Placement = Placement;
-                Popup.PlacementConstraintAdjustment =
-                    PopupPositioning.PopupPositionerConstraintAdjustment.SlideX |
-                    PopupPositioning.PopupPositionerConstraintAdjustment.SlideY;
```

Comment:
You are changing the default value. Now it will be:

https://github.com/AvaloniaUI/Avalonia/blob/9cd8dda5206ffbcb6705ada2ae8d3753aaaf9caf/src/Avalonia.Controls/Primitives/Popup.cs#L55-L60

Is this correct?

Reply:
That's exactly what I am not sure about and I'll let the team decide. The idea back then was to align Popup un Flyout to behave similar. But noone found the root cause they don't do it. Maybe due ro lack of time. So now we have to decide if current behavior was a bug, then new behavior should not be a breaking change but a fix. Otherwise old behavior was by design, then I'll change defaults to old values.

Reply:
My personal opinion is that they should behave the same by default, as done in this PR. (From an end user's perspective, there's no visible difference between a popup and a flyout.)

Reply:
Yes, that probably the right decision.

Note that the difference is probably because Popup followed WPF originally and the first Flyout implementation attempted to follow UWP. The positioning likely differs between them. It may make sense to use the newer UWP positioning settings if that is the case but its probably not that important and can be addressed separately if it ever becomes a problem.